### PR TITLE
Updated Working-with-ResultAsync

### PR DESCRIPTION
**Pull request by lauhon**

### CODE REVIEW

File: Working-with-ResultAsync.md [hunk 1]
```diff
@@ -1,11 +1,68 @@
-`todo`
+Result Async can be used to work with Asynchronous Results in an Error safe way.
```
Comment (supermacro) [inline Working-with-ResultAsync.md]: Suggestion / nitpick:

```suggestion
Result Async can be used to work with Asynchronous Results in a typesafe way.
```

File: Working-with-ResultAsync.md [hunk 2]
```diff
@@ -1,11 +1,68 @@
...
+Similarly to `Result<T,E>`, which contains the Result of a synchronous operation that may have failed, you can handle Promises with `ResultAsync<T,E>`.
```
Comment (supermacro) [inline Working-with-ResultAsync.md]:
```suggestion
Similarly to `Result<T,E>`, which contains the Result of a synchronous operation that may have failed, you can represent the result of async operations with `ResultAsync<T,E>`.

`ResultAsync<T, E>` has the exact same API as `Promise<T>`, but with many more methods, and also a dedicated type (`E`) for the error you expect on your async operation.
```

File: Working-with-ResultAsync.md [hunk 3]
```diff
@@ -1,11 +1,68 @@
...
+
+    result.match(
+        user => console.log("Found my user", user),
+        err => console.log(err.message, err.originalError)
+    )
+}
+```
+
+Notice that typically you would not read the result of ResultAsync immediately after wrapping it, as it does not much other than bloating up your function a bit more.
+The gain of using this paradigm really comes to the surface when you start to chaining different `Result` and `ResultAsnyc` together with the `.map` and `.andThen` functions. This enables you to unwrap (with `match`, or `map` & `mapErr`) the Results at a centralized point to outsource the responsibility to react to Errors and Successes.
```
Comment (supermacro) [inline Working-with-ResultAsync.md]:
```suggestion
The benefit of using this paradigm really comes to the surface when you start chaining different `Result` and `ResultAsnyc` together with the `.map` and `.andThen` functions. This enables you to unwrap (with `match`, or `map` & `mapErr`) the Results at a centralized point to outsource the responsibility to react to Errors and Successes.
```

**Review by supermacro — COMMENTED** [review]
Awesome! Just added some suggestions

**Comment by lauhon** [comment]
> Awesome! Just added some suggestions

Happy you are on board :)
Pushed a commit with your suggestions. 

I will add 1 sentence about the Error that pops up when functions are marked with `async`

Are there any other "big thoughts" or concepts that would make sense on this page from your POV?

**Comment by supermacro** [comment]
Here it is: https://github.com/supermacro/neverthrow/wiki/Working-with-ResultAsync

**Comment by supermacro** [comment]
I skipped the section on linking to the `Intertwine Side Effects Into Neverthrow` doc because that doc doesn't have much in it.

**Pull request closed without merging**